### PR TITLE
Added word2vec script and sample data

### PR DESCRIPTION
**Pull request by ricsuh**

### CODE REVIEW

**Comment by cgreene** [comment]
The first PR! 🎉

File: word2vec.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    """
+    Extracts title + abstracts from Pubtator data. Replaces any instance of a 
+    gene or disease with the Entrez gene ID or MESH ID, respectively. 
+    """
+    def __init__(self, filename):
+        self.filename = filename
+ 
+    def __iter__(self):
+        curr_pmid = curr_title = curr_abstract = curr_total = None
+        for line in open(self.filename):
```
Comment (danich1) [inline word2vec.py]:
```suggestion
        for line in open(self.filename, 'r'):
```
It is good to be explicit on the file mode just so you don't accidentally open a file in writing mode and undo everything.

File: word2vec.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    @param gene_disease_filename: file containing hetionet gene disease pairs
+    @param do_mesh_filename: file containing corresponding doid and mesh ids (because hetnet
+        contains only doids)
+    """
+    random.seed(100)  # reproducibility
+    gene_disease_df = pd.read_csv(gene_disease_filename, sep="\t")
+    do_mesh_df = pd.read_csv(do_mesh_filename, sep="\t")
+
+    # create doid-mesh list
+    do_mesh_pairs = dict(zip(do_mesh_df.doid_code, "MESH:"+do_mesh_df.mesh_id))
```
Comment (danich1) [inline word2vec.py]: Would make your code easier to read if you convert long lines into lines with spaces:
```python
do_mesh_pairs = dict(zip(
     do_mesh_df.doid_code,
     "MESH:"+do_mesh_df.mesh_id
))
```

File: word2vec.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    gene_disease_df["mesh_id"] = gene_disease_df["doid_id"].replace(do_mesh_pairs) 
+    # remove rows that don't have a DOID-MESH id mapping
+    gene_disease_df = gene_disease_df[~gene_disease_df.mesh_id.str.contains("DOID:")]
```
Comment (danich1) [inline word2vec.py]: Side note. You can also do this (dont have to change but just a pandas tip):

```python
gene_disease_df = gene_disease_df.query("~mesh_id.str.contains('DOID:')")
```

File: word2vec.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    # get positive pairs
+    positive_pairs = gene_disease_df[["mesh_id", "entrez_gene_id"]].values.tolist()
```
Comment (danich1) [inline word2vec.py]: White space tip:

```python
positive_pairs = (
   gene_disease_df
   [["mesh_id", "entrez_gene_id"]]
   .values
   .tolist()
)
```

File: word2vec.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    
+    # randomize pairings to create negative pairs
+    gene_disease_df["random_gene"] = random.sample(gene_disease_df["entrez_gene_id"].values.tolist(), len(gene_disease_df["entrez_gene_id"].values.tolist()))
```
Comment (danich1) [inline word2vec.py]: Change to:

```python
gene_disease_df["random_gene"] = (
   random.sample(
      gene_disease_df["entrez_gene_id"].values.tolist(),
       len(gene_disease_df["entrez_gene_id"].values.tolist())
   )
)

File: word2vec.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    """
+    Computes cosine similarity between gene and disease if both exist in the word2vec vocabulary.
+    @param model: the trained word2vec model
+    @param pairs: all gene-disease pairs to be tested
+    """
+    similarity_scores_df = pd.DataFrame(columns=["disease", "gene", "class", "score"]) 
+    for pair in pairs:
+        if all(str(vocab) in model.wv.vocab for vocab in pair[:2]):
+            score = model.wv.similarity(str(pair[0]), str(pair[1]))
+            new_row = {"disease": pair[0], "gene": pair[1], "class": pair[2], "score": score}
```
Comment (danich1) [inline word2vec.py]: Change to:

```python
new_row = {
      "disease": pair[0], 
      "gene": pair[1],
      "class": pair[2], 
      "score": score
}
```
```

File: word2vec.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    predictions = np.array(df[["score"]].values.tolist())
+    return(roc_curve(labels, predictions))
+
+
+if __name__ == "__main__":
+    base = os.path.abspath(os.getcwd())
+    # only need the 3 lines below to run on pmacs cluster
+    sentences = Sentences(os.path.join(base, "data/testdata_pubtator_central_export.pubtator"))
+    word2vec = create_word2vec(sentences)
+    pairs = get_gene_disease_pairs(os.path.join(base, "data/hetnet_gene_disease_pairs.tsv"), os.path.join(base, "data/DO-slim-to-mesh.tsv"))
```
Comment (danich1) [inline word2vec.py]: White space change as above.

**Review by danich1 — APPROVED** [review]
Congrats on your first pull request. 🎉 🎉 

I approve provided that you make the suggested changes above before merging!

**Pull request merged**